src/Parser/Core/Modules/StatTracker.js

```diff
@@ -252,28 +252,55 @@ class StatTracker extends Analyzer {
   }
 
   /*
+   * Helpers to get stat percentage from an arbitrary rating.
+   */
+   getCritPercentage(rating) {
+     return this.baseCritPercentage + (rating / this.critRatingPerPercent);
```

Comment:
I'm not sure about including the base in this as when just reading the method name I'd expect it to do just rating / ratingPerOnePercent and the base would come as a surprise. How about moving this to the methods below?

Reply:
It made sense to me to be able to get a stat percentage given an arbitrary rating, and you can see I actually use the `getHastePercentage` function in the Haste module. Which methods below are you suggesting I move to? I'm not sure what you're recommending here.

Reply:
current* methods

Reply:
I'm still not sure what you mean... are you saying something like `get currentCritPercentage(rating = this.currentCritRating) {`? That doesn't work because `get` methods can't have any arguements.

Reply:
I guess it doesn't matter much as I've been meaning to suggest some other changes to the stat handling that should also make this easier to solve.

Let's focus on crit since it has all the interesting bits.

Right now we have:
- startingCritRating - rating: the crit at pull including any buffs a player had then (base + gear + rating buffs)
- currentCritRating - rating: base + gear + rating buffs
- baseCritPercentage - percentage: base
- critRatingPerPercent - rating: rating per one percent
- getCritPercentage(value) - percentage: base + value
- currentCritPercentage - percentage: base + gear + rating buffs

How about changing this to:
- baseCritRating or baseCritPercentage (rating would make things simpler so we wouldn't have to mix percentage and rating behaviors more than once)
- gearCritRating
- buffCritRating
- currentCritRating - total current rating: base + gear + buff
- critRatingPerPercent - rating: rating per one percent
- getCritPercentage(value) - percentage: value / critRatingPerPercent
- currentCritPercentage - currentCritRating + crit percentage buffs

In this example I consider Haste to behave the same, since Haste does `current * (1 + gain) + gain` I don't think we could get a meaningful `buffCritPercentage` property so I left it out for all stats.

Reply:
I made this PR small because I was trying to connect StatTracker to Haste *without* an involved refactor. I've reverted the StatTracker so there are no longer getStatPercentage(value) functions, and I now use the existing suite to get the same behavior in Haste.

If you think StatTracker's API could be improved, you're welcome to open a PR for it.